cpp/oneapi/dal/io/detail/load_graph.hpp

```diff
@@ -53,126 +54,127 @@ edge_list<std::int32_t> load_edge_list(const std::string &name) {
 
 template <typename Graph>
 void convert_to_csr_impl(const edge_list<vertex_type<Graph>> &edges, Graph &g) {
-    auto layout = oneapi::dal::preview::detail::get_impl(g);
-    using vertex_t = typename Graph::vertex_type;
-    using vector_vertex_t = typename Graph::vertex_set;
-    using vector_edge_t = typename Graph::edge_set;
-    using allocator_t = typename Graph::allocator_type;
-    using vertex_size_t = typename Graph::vertex_size_type;
-
     if (edges.size() == 0) {
-        layout->_vertex_count = 0;
-        layout->_edge_count = 0;
         throw invalid_argument("Empty edge list");
     }
 
-    vertex_t max_node_id = edges[0].first;
+    using vertex_t = typename Graph::vertex_type;
+    using edge_t = typename Graph::edge_type;
+    using vector_vertex_t = typename Graph::vertex_set;
+    using atomic_t = typename daal::services::Atomic<vertex_t>;
```

Comment:
We do not use old interfaces directly in new ones. Moreover, you include old interfaces in public headers - it makes the direct dependency on user side to old DAAL headers. We hide such things in the "interoperability" layer: please see `cpp/oneapi/dal/backend/interop` folder. Such folder can be created for your functionality too, e.g. in `cpp/oneapi/dal/io/backend/interop`. This folder shall contain a wrapper for `daal::services::Atomic`.

Reply:
I'll create a task to do move atomic into oneapi in a proper way.